be/src/vec/aggregate_functions/aggregate_function_linear_histogram.h

```diff
@@ -0,0 +1,241 @@
+        }
+    }
+
+    // insert_result_into
+    void insert_result_into(IColumn& to) const {
+        rapidjson::Document doc;
+        doc.SetObject();
+        rapidjson::Document::AllocatorType& allocator = doc.GetAllocator();
+
+        unsigned num_buckets =
+                buckets.empty() ? 0 : buckets.rbegin()->first - buckets.begin()->first + 1;
+        doc.AddMember("num_buckets", num_buckets, allocator);
+
+        rapidjson::Value bucket_arr(rapidjson::kArrayType);
+        bucket_arr.Reserve(num_buckets, allocator);
+
+        int32_t idx = buckets.begin()->first;
+        double left = buckets.begin()->first * interval + offset;
+        size_t count = 0;
+        size_t acc_count = 0;
+
+        for (const auto& [key, count_] : buckets) {
+            for (; idx <= key; ++idx) {
+                rapidjson::Value bucket_json(rapidjson::kObjectType);
+                bucket_json.AddMember("lower", left, allocator);
+                left += interval;
+                bucket_json.AddMember("upper", left, allocator);
+                count = (idx == key) ? count_ : 0;
+                bucket_json.AddMember("count", count, allocator);
+                acc_count += count;
+                bucket_json.AddMember("acc_count", acc_count, allocator);
+
+                bucket_arr.PushBack(bucket_json, allocator);
+            }
+        }
+
+        doc.AddMember("buckets", bucket_arr, allocator);
+
+        rapidjson::StringBuffer buffer;
+        rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+        doc.Accept(writer);
+
+        auto& column = assert_cast<ColumnString&>(to);
+        column.insert_data(buffer.GetString(), buffer.GetSize());
+    }
+};
+
+template <typename T, typename Data, bool has_offset>
+class AggregateFunctionLinearHistogram final
+        : public IAggregateFunctionDataHelper<
+                  Data, AggregateFunctionLinearHistogram<T, Data, has_offset>> {
+public:
+    using ColVecType = ColumnVectorOrDecimal<T>;
+
+    AggregateFunctionLinearHistogram(const DataTypes& argument_types_)
+            : IAggregateFunctionDataHelper<Data,
+                                           AggregateFunctionLinearHistogram<T, Data, has_offset>>(
+                      argument_types_),
+              scale(get_decimal_scale(*argument_types_[0])) {}
+
+    std::string get_name() const override { return "linear_histogram"; }
+
+    DataTypePtr get_return_type() const override { return std::make_shared<DataTypeString>(); }
+
+    void add(AggregateDataPtr __restrict place, const IColumn** columns, ssize_t row_num,
+             Arena* arena) const override {
+        double interval =
+                assert_cast<const ColumnFloat64&, TypeCheckOnRelease::DISABLE>(*columns[1])
+                        .get_data()[row_num];
+        if (interval <= 0) {
+            throw doris::Exception(ErrorCode::INVALID_ARGUMENT, "Invalid interval {}, row_num {}",
+                                   interval, row_num);
```

Comment:
Add more info for `interval` would be better, i.e., it should be larger than 0.

Reply:
Perhaps this will alert the user that this parameter is set incorrectly.